packages/trace-viewer/src/ui/sourceTab.tsx

```diff
@@ -66,7 +66,9 @@ export const SourceTab: React.FunctionComponent<{
     highlight.push({ line: targetLine, type: 'running' });
 
     // After the source update, but before the test run, don't trust the cache.
-    if (source.content === undefined || shouldUseFallback) {
+    if (fallbackLocation?.source?.content !== undefined) {
+      source.content = fallbackLocation.source.content;
```

Comment:
I don't quite understand this one. Do we show the fallback source even when we have the real source content?

Reply:
Yes, fallbackLocation is a priority.